src/lib/AppLauncherFactory.ts

```diff
@@ -9,14 +15,15 @@ import { AppLaunchStrategyFactory, StrategyEnum } from './AppLaunchStrategyFacto
 * @constructor
 */
 export class AppLauncherFactory {
+  // FIXME: make this class abstract so that it could return one of the sub strategy types
```

Comment:
as I said in this comment, the AppLauncherFactory should be more a superclass of all the *LaunchStrategy so that it could return the right concrete object depending on assigned strategy :) 

In order to make the compiler happy I've added the wildcard field `[x: string]: any` to the class which satisfies the different implementation and the missing fields/methods.